BootloaderCommonPkg/Include/Guid/PayloadKeyHashGuid.h

```diff
@@ -14,14 +14,15 @@
 ///
 extern EFI_GUID gPayloadKeyHashGuid;
 
-#ifndef    SHA256_DIGEST_SIZE
-#define    SHA256_DIGEST_SIZE   32
+#ifndef    HASH_DIGEST_MAX
+#define    HASH_DIGEST_MAX   64
 #endif
 #define    MAX_KEY_DIGEST_COUNT 3
 
 typedef struct {
   UINT32               ComponentType;
-  UINT8                Digest[SHA256_DIGEST_SIZE];
+  UINT8                Digest[HASH_DIGEST_MAX];
+  UINT8                HashType;
```

Comment:
I think it is better to make each HASH_ITEM DWORD aligned. 

Also might be good to consider KeyUsage to support using multiple keys to sign a component.
 UINT8                ComponentType;
 UINT8                HashAlgo;
 UINT8                Reserved[2];
 UINT32              KeyUsage;
 UINT8                Digest[SHA256_DIGEST_SIZE];	  UINT8                Digest[HASH_DIGEST_MAX];

Reply:
Yes, Hash item size would be for SHA512 (64 bits).
As this is defined specific to hash store, I believe  we could add supported key usage support should be global PCD.

Also, I am considering to add support for multiple hashes(as sha256, sha384) for components inconsideration of measured boot to extend multiple PCR's.

UINT8 ComponentType;
UINT8 HashAlgoBitMap;
UINT8 Reserved[2];
UINT8 Digest_256[SHA256_DIGEST_SIZE]; 
UINT8 Digest_384[HASH_DIGEST_MAX];

Reply:
@Maurice : Your comment w.r.t KeyUsage applies to the hashes stored in HashStore too. Isn't it?

Reply:
yes, it does. 
BTW, we should consider size as well.  It is in Stage1A, so size is critical.  To better optimize it, maybe we can use a length field in the structure to indicate hash size.  It can poentially support various hash length then.  How about change Reserved to DigestLength ?

Reply:
@mauricema Could you let know how could to interpret the KeyUsage?

Reply:
In curtain use, it is required to accept multiple keys to sign a blob.  KeyUsage can help indicate that.  It is very similar to the boot guard key manifest.  Please take that as a reference.

Reply:
Thanks Maurice.
Updated data structures in payloadkeyhash,

typedef struct {
  UINT32 ComponentType;
  UINT8 HashAlgo;
  UINT8 Reserved[2];
  UINT8 KeyUsage;
  UINT8 Digest[HASH_INDEX_MAX_NUM];
} KEY_HASH_ITEM;

typedef struct {
  UINT8                Revision;
  UINT8                Reserved0[3];
  UINT32               DigestCount;
  KEY_HASH_ITEM        KeyHash[0];
} PAYLOAD_KEY_HASH;

Reply:
After further discussion,  the following is perfered
typedef struct {
UINT32  Usage;
UINT8    HashAlg;
UINT8    Reserved;
UINT16  DigestLen;
UINT8    Digest[0];
} KEY_HASH_ITEM;

Reply:
We could keep fixed  digest hash to PayloadKeyHashHob where as the HashStore in storage (hashstore.h) can be maintained variable length.

Reply:
If possible, let us use same structure for both storage and HOB so that code can be reused.

Reply:
Yeah, i would maintain same structure for both Storage and HOB